plugins/UFM_NDT_Plugin/ufm_sim_web_service/topo_diff/ndt_infra.py

```diff
@@ -702,3 +745,95 @@ def verify_fix_json_list_file(json_file_to_check):
                 logging.error("Failed to fix json file %s (]] issue)." % json_file_to_check)
                 return False
     return True
+
+def update_cv_credentials(cred_file_path, cv_address, cv_username, cv_password):
+    '''
+    Save credentials encrypted
+    :param cred_file_path
+    :param cv_username:
+    :param cv_password:
+    '''
+    store_credentials = False
+    if not cv_username or not cv_password:
+        logging.error("Failed to save CV server credentials. Username and password do not have values.")
+        return store_credentials
+    enc_cv_username = encrypt(cv_username)
+    enc_cv_password = encrypt(cv_password)
+    # write decrypted values to the file
+    try:
+        f=open(cred_file_path,'wb')
+        f.write(enc_cv_username)
+        f.write(b'\n')
+        f.write(enc_cv_password)
+        f.write(b'\n')
+        store_credentials = True
+    except Exception as e:
+        logging.error("Failed to save encrypted credentials to file %s: $s." % (cred_file_path, e))
+    finally:
+        f.close()
+    return store_credentials
+
+def read_cv_credentials(cred_file_path):
+    '''
+    Save credentials encrypted
+    :param cv_username:
+    :param cv_password:
+    '''
+    # read values from file
+    cv_username = None
+    cv_password = None
+    try:
+        f=open(cred_file_path,'rb')
+        credentials = f.readlines()
+        # should be two lines - username and passsword
+        if len(credentials) != 2:
+            logging.error("Failed to read credentials from file %s." % cred_file_path)
+        else:
+            enc_cv_username = credentials[0].strip()
+            enc_cv_password = credentials[1].strip()
+            # decrypt values
+            cv_username = (decrypt(enc_cv_username)).decode('utf-8')
+            cv_password = (decrypt(enc_cv_password)).decode('utf-8')
+    except Exception as e:
+        logging.error("Failed to read credentials from file %s: %s." % (cred_file_path, e))
+    finally:
+        f.close()
+    return cv_username, cv_password
+
+def update_cv_host_in_config_file(config_file_name, cv_address, cv_port):
+    '''
+    Update config file with cv server addres and port (if need port) - if port
+    different from localhost or 127.0.0.1
+    :param config_file_name
+    :param cv_address:
+    :param cv_port:
+    '''
+    # use bash command to update config file - not to use config user
+    # sed -i -e 's/^cable_validation_server_addr\s*=\s*.*/cable_validation_server_addr=127.0.0.1/' /config/ndt.conf
+    # sed -i -e 's/^cable_validation_request_port\s*=\s*.*/cable_validation_request_port=8633/' /config/ndt.conf
+    if not os.path.isfile(config_file_name):
+        logging.error("Config file %s not exist. Failed to update with CV host name and port." % config_file_name)
+        return False
+    if cv_address:
+        upd_cmd = UPDATE_CONFIG_CMD % ("cable_validation_server_addr",
```

Comment:
why are you using bash commands to update the NDT configuration instead of using the standard python config parser?

Reply:
BASH enough for our needs.
Standard config parcser removes all the comments from cfg file.